handlers/categoryHandler.js

```diff
@@ -0,0 +1,33 @@
+const Category = require('../model/categoryModel');
+
+// get all categories
+const allCategories = async (req,res) => {
+	const categoryLists = await Category.find();
+
+	if(!categoryLists){
+		return res.status(500).json({success:false})
+	}
+	res.status(200).send(categoryLists)
+}
+
+// category creation
+const categoryCreation = async (req,res) => {
+	const { name, icon, color, image } = req.body;
+
+	try {
+		const category = new Category({
+			name, icon, image, color
+		})
+
+		await category
+			.save()
+			.then((createdCategory) => {
+				res.status(201).json({createdCategory})
+			})
+		
```

Comment:
Please check how to use `async` and `await`. already the await resolves your promise why use `then` on the promise ?. The await resolves your promise and return the value, if there is an error then you already have a `try` clause to catch the error which is perfect, and to handle success case all you have to do is write your success code at the next line. do this instead
```suggestion
		const createdCategory = await category.save();
	        res.status(201).json({createdCategory});
```